src/Microsoft.Health.Fhir.Shared.Client/FhirClient.cs

```diff
@@ -81,31 +97,30 @@ public class FhirClient
             }
 
             _mediaType = MediaTypeWithQualityHeaderValue.Parse(_contentType);
-            SetupAuthenticationAsync(clientApplication, user).GetAwaiter().GetResult();
+
+            SetupSecurity();
         }
 
         public ResourceFormat Format { get; }
 
-        public (bool SecurityEnabled, string AuthorizeUrl, string TokenUrl) SecuritySettings { get; }
+        public bool SecurityEnabled { get; private set; }
```

Comment:
Doesn't have to be in this PR, but I feel like we should separate the client and authentication logic at some point.

Reply:
I liked the idea of exposing the `SetupSecurity` call to the using code to do, so I exposed it as the method `ConfigureSecurityOptions`. I also made the `SecurityEnabled` option a `bool?` so it could be known if the method had been called or not.